keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -117,16 +119,16 @@ data class ClusterSpec(
   private val _defaults: ServerGroupSpec,
   @JsonInclude(NON_EMPTY)
   val overrides: Map<String, ServerGroupSpec> = emptyMap(),
-  @JsonIgnore
-  override val deliveryArtifact: DeliveryArtifact? = null,
-  @JsonIgnore
+  // Custom backing field for artifactName, used by resolvers
+  private val _artifactName: String? = null,
```

Comment:
This allows resolvers to write the property while still giving us the ability to provide a custom getter below which attempts to find the artifact info _before_ resolvers have run.